aoc2018/src/main/scala/day10.scala

```diff
@@ -34,11 +45,11 @@ object day10 extends App:
   private val start1: Long =
     System.currentTimeMillis
 
-  private val input: Vector[Star] =
+  private val input: Vector[StarVector] =
 
-    val parser: String => Star = {
-      case s"position=<${x}, ${y}> velocity=<${vx}, ${vy}>" => Star(Point(x.strip().toInt, y.strip().toInt),
-                                                                    vx.strip().toInt, vy.strip().toInt)
+    val parser: String => StarVector = {
+      case s"position=<${x}, ${y}> velocity=<${vx}, ${vy}>" =>
+        StarVector(Point(x.strip().toInt, y.strip().toInt), Point(vx.strip().toInt, vy.strip().toInt))
```

Comment:
preferred style is to not explicitly call `()` on function values - or even side effecting methods these days :)

( during scala 2 days explicit calling was used to signed side effects - which was good practice to signal to other programmers that the method/function was side effecting - but in purely functional codebases side effects are more easily signalled by they code base's function return type being `IO[A]`, `ZIO[R,E,A]`, or even `Future[A]` and the practice was dropped.. ofc this preference is "personal style and thus opinionated" but preferred nonetheless ;)